pulsar/consumer.go

```diff
@@ -181,6 +184,14 @@ type Consumer interface {
 	// ReconsumeLater mark a message for redelivery after custom delay
 	ReconsumeLater(msg Message, delay time.Duration)
 
+	// Reconsume a message
+	ReconsumeLaterLevel(message Message, options ReconsumeOptions) error
+
+	// Reconsume a message Async
+	ReconsumeLaterLevelAsync(message Message,
+		options ReconsumeOptions,
+		callback func(MessageID, *ProducerMessage, error))
```

Comment:
Why is there a ProducerMessage in the callback?

Reply:
When doing message sending, we need it to process the field information such as Payload and Properties

Reply:
```
producerMsg := &ProducerMessage{
		Payload:             msg.Payload(),
		Properties:          propertiesMap,
		ReplicationClusters: msg.(*message).replicationClusters,
	}
```